lib/globus/client/identity.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'faraday'
+
+module Globus
+  class Client
+    # Lookup of a Globus identity ID
+    class Identity
+      def initialize(token)
+        @token = token
+      end
+
+      attr_reader :token
+
+      def connection
+        Faraday.new(
+          url: Settings.globus.auth_url
+        )
+      end
+
+      def lookup_identity
+        id_endpoint = '/v2/api/identities'
+        connection.get(id_endpoint) do |req|
+          req.params['usernames'] = @email
+          req.headers['Authorization'] = "Bearer #{token}"
+        end
+      end
+
+      def get_identity_id(email)
+        @email = email
+        raise StandardError, "Identity #{email} should be in form of: sunet@stanford.edu." unless valid?(@email)
+
+        response = lookup_identity
+        UnexpectedResponse.call(response) unless response.success?
+
+        data = JSON.parse(response.body)
+        extract_id(data)
+      end
+
+      private
+
+      def valid?(email)
+        return true if email.include? '@stanford.edu'
+
```

Comment:
suggest `email.end_with? '@stanford.edu'` to be more precise?

Reply:
Thanks for that tip. Made that change.